libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,202 @@
+                  return this.getProfiles(d.controlName || '').map(
+                    (p) => p.service
+                  );
+                }
+              },
+              userImpacts: {
+                transformer: (d: ControlScore) => {
+                  // return userImpacts from the profile document where its id matches the controlName
+                  return this.getProfiles(d.controlName || '')
+                    .filter((p) => p.userImpact !== undefined)
+                    .map((p) => p.userImpact);
+                }
+              }
+            },
+            source_location: {},
+            code: {
+              transformer: (
+                d: ControlScore & {implementationStatus: string}
+              ) => {
+                const profiles = this.getProfiles(d.controlName || '');
+
+                if (profiles?.length === 0) {
+                  return d.implementationStatus;
+                }
+              }
+            },
+            results: [
+              {
+                status: {
+                  transformer: (
+                    d: ControlScore & {scoreInPercentage: number}
+                  ) => {
+                    if (d.scoreInPercentage === 100) {
+                      return ExecJSON.ControlResultStatus.Passed;
+                    }
+
+                    const knownMaxScores = this.getProfiles(
+                      d.controlName || ''
+                    ).map((p) => p.maxScore || 0);
+
+                    const highMaxScore = Math.max(...knownMaxScores);
+
+                    if (knownMaxScores.length === 0) {
+                      // no Profile found matching the controlName
+                      return ExecJSON.ControlResultStatus.Failed;
+                    } else if (d.score === undefined) {
+                      return ExecJSON.ControlResultStatus.Error;
+                    } else if (d.score === highMaxScore) {
+                      return ExecJSON.ControlResultStatus.Passed;
+                    } else {
+                      return ExecJSON.ControlResultStatus.Failed;
+                    }
+                  }
+                },
+                code_desc: {
+                  transformer: (
+                    d: ControlScore & {implementationStatus: string}
+                  ) => {
+                    const remediations = this.getProfiles(d.controlName || '')
+                      .filter((p) => p.remediation !== undefined)
+                      .map((p) => p.remediation);
+
+                    if (remediations.length > 0) {
+                      return remediations.join('\n\n');
+                    }
+                  }
+                },
+                start_time: {transformer: () => this.data.createdDateTime}
+              }
+            ]
+          }
+        ],
+        sha256: ''
+      }
+    ],
+    passthrough: {
+      profiles: {
+        value: {
+          tramsformer: () => this.profiles
+        }
+      },
```

Comment:
@charleshu-8, @Amndeep7 , 
Thoughts on passing `this.profiles` through?  

* ```Property 'profiles' is used before its initialization.ts(2729)```
* Also, transformers don't seem to get executed within `passthrough`

Reply:
Pushed up a manual fix of passthrough to give you an idea of what it looks like in the other mappers.